dev-libs/optix/optix-7.3.0.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake cuda eapi8-dosym
+
+DESCRIPTION="NVIDIA Ray Tracing Engine"
+HOMEPAGE="https://developer.nvidia.com/optix"
+SRC_URI="NVIDIA-OptiX-SDK-${PV}-linux64-x86_64.sh"
+S="${WORKDIR}"
+
+LICENSE="OptiX-SDK"
+SLOT="0/7"
+KEYWORDS="~amd64"
+IUSE="samples"
```

Comment:
> Recall you mentioned this mattered for blender, is [samples] something blender may end up depending on? If not leave this as-is, but otherwise it may make sense to enable by default.

That's what I was thinking of. I didn't want to build the samples by default, but leave them like they are currently and use the use-dependency for `[samples]` in blender.

> ebuild also isn't doing the gcc check, but think most NVCC ebuilds have given up on checking this except for tensorflow having a:

Good idea to add a gcc check here. Don't have much experience with cuda build either. It's something I'd like to improve in Gentoo. For example, cmake uses the `{compute,sm}_60` GPU by default, if not passed with the `CUDA_NVCC_FLAGS` option. We could use this to pass the desired GPU to cmake, but there's no reliable way in `cuda.eclass` to select the GPU. We only have user settable `NVCCFLAGS` in the eclass, which *can* contain GPU settings as well. Not sure, however, how many users make use of this.